Added a proxy API for token refresh

alexeykazakov opened the pull request:
This is a proxy API for token refresh in case we decide to go with a proxy instead of using a public Keycloak client.

alexeykazakov commented:
Example: 

POST /api/login/refresh

payload: {"refresh_token": "eyJh..."}

Response:
```

{
   "token": {
       "access_token": "eyJh...",
       "expires_in": 1800,
       "not-before-policy": 0,
       "refresh_expires_in": 1800,
       "refresh_token": "eyJhb...",
       "token_type": "bearer"
   }
}
```

aslakknutsen commented on login.go:
```diff
@@ -40,6 +40,64 @@ func (c *LoginController) Authorize(ctx *app.AuthorizeLoginContext) error {
+		// OK
+	case 401:
+		jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrUnauthorized(readBody(res.Body)))
+		return ctx.Unauthorized(jerrors)
```
Use the same format for error handling as below

alexeykazakov commented:
@aslakknutsen I fixed the error and added tests. Now this PR is ready to be merged if we think it's a better approach to use a api proxy for refresh instead of a public client.

hectorj2f commented on login.go:
```diff
@@ -40,6 +40,63 @@ func (c *LoginController) Authorize(ctx *app.AuthorizeLoginContext) error {
+	case 400:
+		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError(readBody(res.Body), nil))
+	default:
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(res.Status+""+readBody(res.Body)))
```
Shouldn't it be a whitespace in between ? like u did in line 64

alexeykazakov replied:
oops.. You are right :)

hectorj2f commented on login.go:
```diff
@@ -40,6 +40,63 @@ func (c *LoginController) Authorize(ctx *app.AuthorizeLoginContext) error {
+	var token app.TokenData
+	err := json.Unmarshal([]byte(jsonString), &token)
+	if err != nil {
+		return nil, jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(fmt.Sprintf("Error when unmarshal json with access token %s ", jsonString)+err.Error()))
```
As a future convention, we might want to avoid capitalized error messages when possible. https://github.com/golang/go/wiki/CodeReviewComments#error-strings

alexeykazakov replied:
Makes sense. Good to know. Thanks.
